features/src/autogluon/features/generators/drop_duplicates.py

```diff
@@ -132,6 +132,7 @@ def _drop_duplicate_features_categorical(cls, X: DataFrame, keep: Union[str, boo
             # Converts [5, 'a', np.nan, 5] to [0, 1, 2, 0], these would be considered duplicates since they carry the same information.
 
             # Have to convert to object dtype because category dtype for unknown reasons will refuse to replace NaNs.
+            # TODO: Fix FutureWarning
             X_cur = X[features_to_check].astype("object").replace(mapping_features_val_dict_cur).astype(np.int64)
```

Comment:
@Innixma this line raises `FutureWarning`:
```
  /local/home/shchuro/workspace/autogluon/features/src/autogluon/features/generators/drop_duplicates.py:135: FutureWarning: Downcasting behavior in `r
eplace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-
in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
    X_cur = X[features_to_check].astype("object").replace(mapping_features_val_dict_cur).astype(np.int64)
```
Do you have any recommendations on how to fix it?

Reply:
will investigate and send PR